app/src/main/java/com/keylesspalace/tusky/appstore/CacheUpdater.kt

```diff
@@ -26,12 +26,35 @@ class CacheUpdater @Inject constructor(
             eventHub.events.collect { event ->
                 val accountId = accountManager.activeAccount?.id ?: return@collect
                 when (event) {
-                    is FavoriteEvent ->
-                        timelineDao.setFavourited(accountId, event.statusId, event.favourite)
-                    is ReblogEvent ->
-                        timelineDao.setReblogged(accountId, event.statusId, event.reblog)
-                    is BookmarkEvent ->
-                        timelineDao.setBookmarked(accountId, event.statusId, event.bookmark)
+                    is StatusChangedEvent -> {
```

Comment:
Just an observation: "status changed" has much less (specific) information than for example "fav clicked". I would immediately ask "what changed?".

I see that these events are normally not used in their specific sense. But maybe they could?

Is it possible to have an event hierarchy? BookmarkEvent extends StatusChangedEvent? Or is that bad style?

Reply:
That is the idea that there is less info. We need to update everything anyway. If the post has been bookmarked, the fav count could have changed as well. Or it was favourited on another device.